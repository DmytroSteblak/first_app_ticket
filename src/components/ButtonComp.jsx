import React from 'react';

import { Button } from 'react-bootstrap';
import { Btn } from '../styled/Button-styled';
import { Link } from 'react-router-dom';

const ButtonComp = ({children, router, onClick}) => (
    <Btn>
        <Link to={`/${router}`}>
            <Button onClick={() => onClick()}>
                {children};
            </Button>
        </Link>
    </Btn>
);

export default ButtonComp;