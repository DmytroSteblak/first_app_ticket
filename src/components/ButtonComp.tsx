import React from 'react';
import { Button } from 'react-bootstrap';

import { Btn } from '../styled/Button-styled';
import { TypeButton } from '../types';

const ButtonComp: React.FC<TypeButton> = ({ children, onClick }) => (
    <Btn>
        <Button onClick={() => onClick()}>
            { children }
        </Button>
    </Btn>
);

export default ButtonComp;
