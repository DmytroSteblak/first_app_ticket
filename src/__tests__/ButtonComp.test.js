import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ButtonComp from '../components/ButtonComp';

const props = {
    children: 'test text',
    onClick: jest.fn()
}

describe('Button component <ButtonComp/>',function () {
    let wrapper, actionMock;

    beforeEach(() => {
        actionMock = jest.fn();
        wrapper = shallow(<ButtonComp action={actionMock}>Cancel</ButtonComp>);
    });

    it('should render ButtonComp component with props', function () {
        const component = shallow(<ButtonComp {...props} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Test click event', () => {
        const mockCallBack = jest.fn();
        const button = shallow((<ButtonComp onClick={mockCallBack}>Ok!</ButtonComp>));
        button.find('Button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

});