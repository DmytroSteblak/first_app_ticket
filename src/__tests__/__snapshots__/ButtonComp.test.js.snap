// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component <ButtonComp/> should render ButtonComp component with props 1`] = `
<styled.div>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    test text
  </Button>
</styled.div>
`;
